### PR TITLE
Basic Summon Command

### PR DESCRIPTION
**Pull request by bearbin**
Replaces #190

### CODE REVIEW

File: summon.lua [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+	-- 1.10 and below
+	["MinecartChest"] = E_ITEM_CHEST_MINECART,
+	["MinecartFurnace"] = E_ITEM_FURNACE_MINECART,
+	["MinecartHopper"] = E_ITEM_MINECART_WITH_HOPPER,
+	["MinecartRideable"] = E_ITEM_MINECART,
+	["MinecartTNT"] = E_ITEM_MINECART_WITH_TNT
+}
+
+local Mobs =
```
Comment (NiLSPACE) [inline summon.lua]: There is a `cMonster:StringToMobType` method. Wouldn't that work?

File: summon.lua [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	["ThrownEnderpearl"] = cProjectileEntity.pkEnderPearl,
+	["ThrownExpBottle"] = cProjectileEntity.pkExpBottle,
+	["ThrownPotion"] = cProjectileEntity.pkSplashPotion,
+	["WitherSkull"] = cProjectileEntity.pkWitherSkull
+}
+
+function RelativeCommandCoord(a_Split, a_Relative)
+	if string.sub(a_Split, 1, 1) == "~" then
+		local rel = tonumber(string.sub(a_Split, 2, -1))
+		if rel {
```
Comment (NiLSPACE) [inline summon.lua]: Invalid syntax here. Lua uses 

```lua
if rel then
    -- code here
end
```

File: summon.lua (hunk 1 above)
Comment (bearbin) [inline summon.lua]: I'm not sure why @mathiascode used the tables originally, as it's his code. I shall change it to use StringToMobType

**Comment by bearbin** [comment]
I made the change. Perhaps a similar function for projectiles should be implemented in due course.

File: summon.lua (hunk 1 above)
Comment (mathiascode) [inline summon.lua]: I used tables, because the mob strings in Cuberite aren't identical to Vanilla

Comment (bearbin) [inline summon.lua]: I will change it back to tables - maintaining vanilla compatibility is relatively important.

**Comment by mathiascode** [comment]
MobTypeToVanillaName might work, but only for versions 1.10 and below. MobTypeToVanillaNBT is for newer vanilla names, but it's not exported to the API.

**Comment by bearbin** [comment]
@NiLSPACE, are you happy with this now that the table issue is cleared up?

**Comment by NiLSPACE** [comment]
Yeah, no problem :)

**Pull request merged**